### PR TITLE
feat: Allow adding components to commands

### PR DESCRIPTION
**Pull request by hydrobeam**
## Summary of Changes

- Allowed adding components to messages. These work for both Slash and Regular commands.
- Added the type `components` to `CodeyCommandDetails` to allow sub-command declarations in `commandDetails` to declare components.
   - These are done through builders functions provided by discord.js directly, no abstraction.
   - **WARNING**: Was forced to use `discord.js v13` names, so #271 will have to be updated.
- **How it works**: If the `commandDetails` contains any components,  then don't execute the command and let the class under `interaction-handlers` do the job,
- `interaction-handlers` is currently pretty sapphire intertwined, but it shouldn't be tough to detangle imo.
- Could easily be used with Buttons, Modals, etc ...  
- See [here](https://www.sapphirejs.dev/docs/Guide/application-commands/interaction-handlers/what-are-they) for an overview of interactions (sapphire docs)

## Motivation and Explanation

Components can do cool things, like implement SelectMenus + other interactions:

https://discordjs.guide/interactions/select-menus.html#building-and-sending-select-menus

## Related Issues

N/A

## Steps to Reproduce

N/A

## Demonstration of Changes

See #304 `/interviewer domain` for a showcase of an implementation

### CODE REVIEW

File: src/codeyCommand.ts [hunk 1]
```diff
@@ -154,13 +165,14 @@ export class CodeyCommandDetails {
...
   codeyCommandResponseType?: CodeyCommandResponseType = CodeyCommandResponseType.STRING;
-
   /** Options for the Codey command */
   options: CodeyCommandOption[] = [];
   /** Subcommands under the CodeyCommand */
   subcommandDetails: { [name: string]: CodeyCommandDetails } = {};
   /** The default subcommand to execute if no subcommand is specified */
   defaultSubcommandDetails?: CodeyCommandDetails;
+  /** Components to be shown in the message, type stolen from discord.js */
+  components?: (MessageActionRow | (Required<BaseMessageComponentOptions> & MessageActionRowOptions))[] = [];
```
Comment (mcpenguin) [inline src/codeyCommand.ts:175]: Extract `MessageActionRow | (Required<BaseMessageComponentOptions> & MessageActionRowOptions))` into a `CodeyComponent` type, and use that here? That way if the component class ever changes, we can just change that.

**Review by mcpenguin — CHANGES_REQUESTED** [review]
This is a little weird, but if it works, it works

**Comment by hydrobeam** [comment]
> This is a little weird, but if it works, it works


Yeah taking another look at this implementation and after having worked on #329, it doesn't really make sense for `components` to be treated any differently from embeds. So `executeCommand` should be used to generate the component, and then the actual managing of the resonse should be done in the `run` method under the exported class in `interaction-handlers`. 

My approach in #304 was to have `interaction-handlers` call back to `executeCommand`, which is weird and doesn't follow the trend established by other commands.  Would be worth introducing an optional `componentHandler` function or something like that to keep the logic centralized.

**Comment by mcpenguin** [comment]
I think a `ComponentHandler`, or even a more generic `Handler` class would be good to centralize the logic, as you mentioned. In the future, as part of desapphirication of the interaction handler, we could employ the observer pattern or something similar (haha CS246 go brrrr) to register listeners for interactions for responding to components. That is outside the scope of this ticket though - I think making a handler is the right way to go here.

**Comment by mcpenguin** [comment]
Closing - PR is not needed anymore

**Pull request closed without merging**